source/config.coffee

```diff
@@ -0,0 +1,34 @@
+import path from "path"
+import fs from "fs/promises"
+import { compile } from "./main"
+import { pathToFileURL } from "url"
+
+export findConfig = ->
+    curr = process.cwd()
+    parent = path.dirname curr
+    while curr isnt parent # root directory (/, C:, etc.)
+        dir = await fs.opendir curr
+        for await entry from dir
+            if entry.name in ['config.civet', '.civetconfig', 'civet.json'] 
+                return path.join curr, entry.name
+        curr = parent
+        parent = path.dirname curr
+    null
+
+export loadConfig = (path) ->
+    config = await fs.readFile path, 'utf8'
+    if path.endsWith '.json'
+        JSON.parse config
+    else
+        js = compile config, js: true
+        
+        tmpPath = path + ".civet-tmp-#{Date.now()}.mjs"
+        await fs.writeFile tmpPath, js
```

Comment:
I wonder if you could share code with `cli.coffee`?

Reply:
Was thinking this too, but we need to get the `export default` from the config file. `cli.coffee` only needs to run it, so it can just start a new node process.

(there is also the option of wrapping the entire CLI with the `.civet` ESM loader so that both are just an `await import`, but node.js loaders are experimental and can change at any time)